src/analyses/base.ml

```diff
@@ -87,14 +87,50 @@ struct
 
   let is_privglob v = GobConfig.get_bool "annotation.int.privglobs" && v.vglob
 
-  let project_val p_opt value is_glob =
-    match GobConfig.get_bool "annotation.int.enabled", is_glob, p_opt with
-    | true, true, _ -> VD.project PU.max_int_precision value
-    | true, false, Some p -> VD.project p value
-    | _ -> value
+  (*This is a bit of a hack to be able to change array domains if a pointer to an array is given as an argument*)
+  (*We have to prevent different domains to be used at the same time for the same array*)
+  (*After a function call, the domain has to be the same as before and we can not depend on the pointers staying the same*)
+  (*-> we determine the arrays a pointer can point to once at the beginning of a function*)
+  (*There surely is a better way, because this means that often the wrong one gets chosen*)
+  module VarMap = Map.Make(CilType.Varinfo)
+  let array_map = ref VarMap.empty
+
+  let add_to_array_map fundec arguments =
+    let rec pointedArrayMap = function
+      | [] -> VarMap.empty
+      | (info,value)::xs -> (
+        match value with
+          | `Address t when hasAttribute "goblint_array_domain" info.vattr -> (
+            let possibleVars = PreValueDomain.AD.to_var_may t in
+            List.fold_left (fun map arr -> VarMap.add arr (info.vattr) map) (pointedArrayMap xs) @@ List.filter (fun info -> isArrayType info.vtype) possibleVars
+          )
+          | _ -> pointedArrayMap xs
+      )
+  in
+  match VarMap.find_opt fundec.svar !array_map with
+      (*We already have something -> do not change it*)
+      | Some _ -> ()
+      | None -> array_map := VarMap.add fundec.svar (pointedArrayMap arguments) !array_map
```

Comment:
This indentation looks wrong.

Reply:
I think this is still the case. Usually, we align as follows

~~~OCaml
match foo with
| Some Bar -> ...
| Some Baz -> ...
| None -> ...
~~~